modules/game/src/main/BinaryFormat.scala

```diff
@@ -82,13 +81,13 @@ object BinaryFormat {
 
   case class clock(since: DateTime) {
 
-    def write(clock: Clock): ByteArray = ByteArray {
-      def time(t: Float) = writeSignedInt24((t * 100).toInt)
-      def timer(seconds: Double) = writeTimer((seconds * 100).toLong)
-      Array(writeClockLimit(clock.limit), writeInt8(clock.increment)) ++
-        time(clock.whiteTime) ++
-        time(clock.blackTime) ++
-        timer(clock.timerOption getOrElse 0d) map (_.toByte)
+    def write(clock: Clock): ByteArray = {
+      ByteArray {
+        Array(writeClockLimit(clock.limitSeconds), writeInt8(clock.incrementSeconds)) ++
+          writeSignedInt24(clock.whiteTime.value) ++
+          writeSignedInt24(clock.blackTime.value) ++
+          (clock.timerOption map { ts => writeTimer(ts.value / 10) } getOrElse Array()) map { _.toByte }
```

Comment:
sounds good.  I have a follow up brewing for this file to optimize some of the serialization.  Will incorporate that idea.

I assume you're saying it's not as typesafe because scala might just decide to return an `Option[AnyVal]`?